source/tutorials/first-steps/nix-language.md

```diff
@@ -2046,11 +2046,13 @@ This example is a (simplified) package declaration from Nixpkgs.
 
 Explanation:
 
-- This expression is a function that takes an attribute set which must have exactly the attributes `lib` and `stdenv`.
+- This expression is a function that takes an attribute set which must have exactly the attributes `lib`, `stdenv`, and `fetchurl`.
```

Comment:
Perhaps it way be worthwhile to mention here why `nixpkgs` expressions don't need to use `...`. (The answer is that `callPackage` uses `builtins.functionArgs` to introspect the function, so it knows what arguments it expects.)

Reply:
While it's true, I don't think we should pollute overwhelmed beginners' minds with additional information. `callPackage` is entirely out of scope here. That bit should be documented in Nixpkgs contributing guides or the part on `callPackage`.

Reply:
For what it's worth, here is my experience as a beginner:

1. Read https://nixos.org/manual/nix/stable/language/index.html
2. Read this tutorial
3. Be confused why `...` is not required here. Was the language spec out of date? I had to ask on the Matrix channel to learn that it works because the caller does introspection.

Reply:
I understand. Still I'd like not to expand the scope here. The `callPackage` pattern is surely worth an introduction of it's own, and we could import and slightly update this article: https://summer.nixos.org/blog/callpackage-a-tool-for-the-lazy/ (it needs a new home anyway) and then link to it the end of this tutorial.